libs/MeshKernel/src/CurvilinearGrid/CurvilinearGridSmoothness.cpp

```diff
@@ -0,0 +1,93 @@
+//---- GPL ---------------------------------------------------------------------
+//
+// Copyright (C)  Stichting Deltares, 2011-2023.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation version 3.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+//
+// contact: delft3d.support@deltares.nl
+// Stichting Deltares
+// P.O. Box 177
+// 2600 MH Delft, The Netherlands
+//
+// All indications and logos of, and references to, "Delft3D" and "Deltares"
+// are registered trademarks of Stichting Deltares, and remain the property of
+// Stichting Deltares. All rights reserved.
+//
+//------------------------------------------------------------------------------
+
+#include <cmath>
+
+#include "MeshKernel/Constants.hpp"
+#include "MeshKernel/Vector.hpp"
+
+#include "MeshKernel/CurvilinearGrid/CurvilinearGridSmoothness.hpp"
+
+void meshkernel::CurvilinearGridSmoothness::Compute(const CurvilinearGrid& grid, const int direction, lin_alg::Matrix<double>& smoothness)
+{
+    smoothness.resize(grid.m_numM, grid.m_numN);
+    smoothness.fill(constants::missing::doubleValue);
+
+    if (direction == 1)
```

Comment:
Can use a switch when enum is in place. Does it make sense to have an `Unknown` enum? in other words, is it needed? if yes, have `case default` and `case SmoothingDirection` bundled and throw an error.

Reply:
I think a switch here make complicates thing too much.